### ISSUE TITLE
Implement `CallBack`

### ISSUE THREAD

**Issue by 5ei74R0** [OtsuKotsu/MLExp#2]
# Implement `CallBack`
### 1. implement as a class used in `experiment.runner.run`
  - `experiment.runner.run` provide too abstract feature (e.g. `metrics: metrics_t`)
    - e.g. early stopping requires you to give specific metric and it is difficult or too complicated to use `metrics: metrics_t` because `metrics_t` has recursive structure so it is difficult to implement the code that is feasible in every case.
    - e.g. learning rate modification needs you to pass optimizer that is passed to training loop function

### 2. implement `callback_holder` as an decorator of training loop function
- callbacks can get more & detailed information naturally
- cannot terminate `run` naturally in cases such as early stopping

**Comment by 5ei74R0** [comment]
#### 1. Implement `nested_dict2dict(nested: nested_dict_t, prefix_connector: str)`
By preparing this function, we can use simple dictionary in the `runner` and we can assume that `run.__call__()` treats only `dict[str, float]` as an metrics holder. it makes it easy to use metrics in callbacks such as early_stopping

**Comment by OtsuKotsu** [comment]
#### 1-alter. Implement operator `[Iterable[str]]` as an access operator of `nested_dict`
